File: solutions/fortran/collatz-conjecture/2/collatz_conjecture.f90

module collatz_conjecture
  implicit none
contains

  integer function steps(i)
    integer, intent(in) :: i
    integer :: num
    num = i
    steps = -1
    do while (num>=1)
      steps = steps + 1  
      if (num==1) then
        exit
      end if
      if (mod(num,2) == 0) then
        num = num / 2
      else
        num = 3 * num + 1
      end if
     
    end do  
    return
  end function

end module

File: solutions/vbnet/collatz-conjecture/1/CollatzConjecture.vb
Imports System

Public Module CollatzConjecture
    Public Function Steps(ByVal number As Integer) As Integer
        Dim s As Integer
        If number <= 0 Then
            Throw New ArgumentOutOfRangeException
        End If
        s = 0
        While number > 1
            s = s + 1
            If number Mod 2 = 0 Then
                number = number / 2
            Else
                number = 3 * number + 1
            End If
        End While
        Return s
    End Function
End Module